libcodechecker/libhandlers/server.py

```diff
@@ -342,6 +369,193 @@ def arg_match(options):
+    elif ret == DBStatus.MISSING:
+        return "Database is missing"
+
+
+def get_user_input(msg):
+    """
+    Get the user input.
+
+    :returns: True/False based on the asnwer from the user
+    """
+    yes = ['y', 'Y', 'yes', 'Yes']
+
+    answer = raw_input(msg)
+    if answer in yes:
+        return True
+    return False
+
+
+def print_prod_status(prod_status):
+    """
+    Print the database statuses for each of the products.
+    """
+
+    header = ['Product endpoint', 'Database status']
+    rows = []
+
+    for k, v in prod_status.items():
+        rows.append([k, handle_connect_ret(v)])
+
+    prod_status = output_formatters.twodim_to_str('table',
+                                                  header,
+                                                  rows,
+                                                  sortby_column_number=0)
+    print(prod_status)
+
+
+def check_product_db_status(cfg_sql_server, context):
+    """
+    Check the products for database statuses.
+
+    :returns: dictionary of product endpoints with database statuses
+    """
+
+    LOG.debug("Checking config database")
+    migration_root = context.run_migration_root
+
+    engine = cfg_sql_server.create_engine()
+    config_session = sessionmaker(bind=engine)
+    sess = config_session()
+
+    products = sess.query(ORMProduct).all()
+
+    prod_status = {}
+    for pd in products:
+        db = database.SQLServer.from_connection_string(pd.connection,
+                                                       RUN_META,
+                                                       migration_root,
+                                                       interactive=False)
+
+        ret = db.connect()
+        prod_status[pd.endpoint] = ret
+
+    sess.commit()
+    sess.close()
+    engine.dispose()
+
+    return prod_status
+
+
+def __db_status_check(cfg_sql_server, context, product_name=None):
+    """
+    Check and print database statuses for the given product.
+    """
+    if not product_name:
+        return 0
+
+    print(product_name)
+
+    prod_statuses = check_product_db_status(cfg_sql_server, context)
+
+    if product_name != 'all':
+        avail = prod_statuses.get(product_name)
+        if not avail:
+            LOG.error("No product was found with this endpoint: " +
+                      str(product_name))
+            return 1
+
+        prod_statuses = {k: v for k, v in prod_statuses.items()
+                         if k == product_name}
+
+    print_prod_status(prod_statuses)
+    return 0
+
+
+def __db_migration(cfg_sql_server, context, product_to_upgrade):
+    """
+    Handle database management.
+    Schema checking and migration.
+    """
+    LOG.info("Migrating schema for " + str(product_to_upgrade))
+    product_name = product_to_upgrade
+
+    prod_statuses = check_product_db_status(cfg_sql_server, context)
+    prod_to_upgrade = []
+
+    if product_name != 'all':
+        avail = prod_statuses.get(product_name)
```

Comment:
You use this piece of code in multiple places: https://github.com/gyorb/codechecker/blob/16b2583583258cf5a5935a01f3385b1c7ac9a388/libcodechecker/libhandlers/server.py#L469-L473
Create a helper function for this.